### ISSUE TITLE
Limit runtime of script execution

### ISSUE THREAD

**Issue by ad-si** [denoland/deno#16461]
Is there a way to limit the runtime of a script execution to x milliseconds like this (or similar)?

```sh
deno run --max-runtime 3000 test.js
```

**Comment by ad-si** [comment]
Yeah, that's what I'm using right now, but it feels really hacky to go this way and there is most certainly some performance overhead with this solution.

**Comment by littledivy** [comment]
There's no built-in `--max-runtime` flag, and a wall-clock kill switch in the CLI is largely outside Deno's scope since the OS/userland already cover it cleanly: `timeout 3 deno run test.js`, or in-script `setTimeout(() => Deno.exit(1), 3000)` / `AbortSignal.timeout(3000)`. Closing as not-planned; the workarounds above are the recommended approach.
